hi-ml-histopathology/testhisto/testhisto/mocks/tiles_datamodule.py

```diff
@@ -0,0 +1,165 @@
+#  ------------------------------------------------------------------------------------------
+#  Copyright (c) Microsoft Corporation. All rights reserved.
+#  Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+#  ------------------------------------------------------------------------------------------
+import os
+import numpy as np
+import pandas as pd
+
+from typing import Any, Tuple
+import torch
+from torchvision.utils import save_image
+
+from health_ml.utils.split_dataset import DatasetSplits
+from histopathology.datamodules.base_module import TilesDataModule
+from histopathology.datasets.base_dataset import TilesDataset
+from testhisto.mocks.base_datamodule import MockHistoDataGenerator
+
+
+class MockTilesDataset(TilesDataset):
+    """Mock and child class of SlidesDataset, to be used for testing purposes.
+    It overrides the following, according to the PANDA cohort settings:
+
+    :param LABEL_COLUMN: CSV column name for tile label set to "slide_isup_grade".
+    :param SPLIT_COLUMN: CSV column name for train/test split (None for MockTiles data).
+    :param N_CLASSES: Number of classes indexed in `LABEL_COLUMN`.
+    """
+
+    LABEL_COLUMN = "slide_isup_grade"
+    SPLIT_COLUMN = None
+    N_CLASSES = 6
+
+
+class MockTilesDataModule(TilesDataModule):
+    """Mock and child class of TilesDataModule, overrides get_splits so that it uses MockTilesDataset."""
+
+    def get_splits(self) -> Tuple[MockTilesDataset, MockTilesDataset, MockTilesDataset]:
+        dataset = MockTilesDataset(self.root_path)
+        splits = DatasetSplits.from_proportions(
+            dataset.dataset_df.reset_index(),
+            proportion_train=0.8,
+            proportion_test=0.1,
+            proportion_val=0.1,
+            subject_column=dataset.TILE_ID_COLUMN,
+            group_column=dataset.SLIDE_ID_COLUMN,
+        )
+        return (
+            MockTilesDataset(self.root_path, dataset_df=splits.train),
+            MockTilesDataset(self.root_path, dataset_df=splits.val),
+            MockTilesDataset(self.root_path, dataset_df=splits.test),
+        )
```

Comment:
This is another place where I would use patching to call `MockTilesDataset` instead of `TilesDataset` - tests should not replicate the logic of the code we want to test